src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -369,6 +370,8 @@ private function createFirewall(ContainerBuilder $container, $id, $firewall, &$a
         // Exception listener
         $exceptionListener = new Reference($this->createExceptionListener($container, $firewall, $id, $configuredEntryPoint ?: $defaultEntryPoint, $firewall['stateless']));
 
+        $container->setAlias(new Alias('security.user_checker.'.$id, false), $firewall['user_checker']);
```

Comment:
It was the only way I was able to get the user checker defined per firewall. I use an Alias because of the following configuration:
```yml
        security:
            firewalls:
                secured_area:
                    pattern: ^/
                    user_checker: app.user_checker
                admin:
                    pattern: ^/admin
                    user_checker: app.admin_user_checker
    
```
When the firewall part is being created, I couldn't get the user checker in there, I can only distinguish them based on`$id` which is the firewall name. By creating an alias per firewall, I can see which is which.

_If you have a better idea, I can see if I can make that work instead._